pkg/config/tokens.go

```diff
@@ -0,0 +1,64 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/coinbase/rosetta-geth-sdk/configuration"
+)
+
+// ReadTokenConfig attempts to read file contents from a filename.
+func ReadTokenConfig(filename string) (string, error) {
+	contents := ""
+	if file, err := os.ReadFile(filename); err == nil {
+		// if the envvar points to a file, read it; otherwise the envvar contents is expected to be JSON
+		contents = string(file)
+	}
+	if contents == "" {
+		contents = "[]"
+	}
+	return contents, nil
+}
+
+// UnmarshalTokenConfig attempts to construct a list of [configuration.Token] from a JSON file.
+func UnmarshalTokenConfig(contents string) ([]configuration.Token, error) {
+	// Try to parse the file as a list of tokens.
+	var payload []configuration.Token
+	if err := json.Unmarshal([]byte(contents), &payload); err == nil {
+		return payload, nil
+	}
+
+	// If this fails, try the backwards-compatible token json format
+	var outer map[string]interface{}
+	if err := json.Unmarshal([]byte(contents), &outer); err == nil {
```

Comment:
This code is adding all tokens in the list, rather than selecting the token list for the configured network.
The legacy token JSON contains multiple configurations for each environment. op-rosetta must then select a configuration based on the configured network/environment. For example, in the case of this legacy token list:
```json
{
  "Mainnet" : {
    "0x4200000000000000000000000000000000000042": true,
    "0xda10009cbd5d07dd0cecc66161fc93d7c9000da1": true,
    "0x94b008aa00579c1307b0ef2c499ad98a8ce58e58": true,
    "0x68f180fcce6836688e9084f035309e29bf0a2095": true,
    "0x7f5c764cbc14f9669b88837ca1490cca17c31607": true
  },
  "Testnet" : {
    "0x4200000000000000000000000000000000000042": true,
    "0xda10009cbd5d07dd0cecc66161fc93d7c9000da1": true,
    "0x853eb4ba5d0ba2b77a0a5329fd2110d5ce149ece": true,
    "0xe0a592353e81a94db6e3226fd4a99f881751776a": true,
    "0x7e07e15d2a87a24492740d16f5bdf58c16db0c4e": true
  }
}
```
either the `Mainnet` or `Testnet` token list should be used, but not both.

Reply:
(we also probably want to case-insensitive when selecting a token list)